### PR TITLE
Fix blind signing and multiple transaction sending for legacy

### PR DESCRIPTION
**Pull request by abenso**

### CODE REVIEW

File: app/src/apdu_handler_legacy.c [hunk 1]
```diff
@@ -288,6 +288,7 @@ bool legacy_process_transfer_chunk(uint32_t rx) {
             if (items != LEGACY_TRANSFER_NUM_ITEMS) {
                 THROW(APDU_CODE_DATA_INVALID);
             }
+            tx_initialized = false;
```
Comment (emmanuelm41) [inline app/src/apdu_handler_legacy.c]: shouldn't this go inside the previous if statement, as it happens at the beginning of this function?

**Review by emmanuelm41 — COMMENTED** [review]
Just a comment to understand better

File: app/src/apdu_handler_legacy.c (hunk 1 above)
Comment (abenso) [reply app/src/apdu_handler_legacy.c]: you are right, we need to reset the flag if an error occurs

**Review by emmanuelm41 — COMMENTED** [review]
Wouldn't it be better if we catch the error with our macro, and set it to false directly there for any error that may happen? I know there is a macro like TRY/CATCH for that

**Comment by abenso** [comment]
I prefer to use the catch block only for error handling and response formatting to avoid mixing concerns

**Review by emmanuelm41 — APPROVED** [review]

**Pull request merged**